azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -398,12 +402,78 @@ public void createContainer(final int executionId) throws ExecutorManagerExcepti
    */
   @Override
   public void deleteContainer(final int executionId) throws ExecutorManagerException {
-    deletePod(executionId);
+    try { // if pod deletion is not successful, the service deletion can still be handled
+      deletePod(executionId);
+    } catch (ExecutorManagerException e) {
+      logger.error("ExecId: {}, Unable to delete Pod in Kubernetes: {}", executionId,
+          e.getMessage());
+    }
     if (isServiceRequired()) {
       deleteService(executionId);
     }
   }
 
+  /**
+   * This method is used to delete invalid containers in batch
+   * @param containerValidity pod validity in milliseconds
+   * @throws ExecutorManagerException
+   */
+  @Override
+  public void deleteContainers(long containerValidity) throws ExecutorManagerException {
```

Comment:
Wondering if we should send an email to Azkaban admins about the list of containers that could not be deleted?

Reply:
I'll address this in a separate PR. Right now logging the error should be sufficient.